src/main/java/seedu/address/logic/commands/MarkAssignmentCommand.java

```diff
@@ -45,14 +48,25 @@ public MarkAssignmentCommand(Name name, Index targetIndex) {
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        List<Assignment> lastShownList = model.getFilteredAssignmentList(name);
+        // Get Person that match the name
+        List<Person> filteredPersonList =
+                model.getFilteredPersonList()
+                        .stream()
+                        .filter(person -> person.isSameName(name))
+                        .collect(Collectors.toList());
 
+        if (filteredPersonList.size() == 0) {
+            throw new CommandException(MESSAGE_INVALID_PERSON_DISPLAYED_NAME);
+        }
+
+        Person selectedPerson = filteredPersonList.get(0);
+        List<Assignment> lastShownList = model.getFilteredAssignmentList(selectedPerson);
         if (lastShownList.size() == 0) {
```

Comment:
in latter version, i try to abstract it in an more OOP way, not sure possible or not since there are lots of  duplication